### PR TITLE
#9663: Fix radius bug in CylinderBufferGeometry

### PR DESCRIPTION
**Pull request by Mugen87**
See #9663.

The zero comparison does not work in some special cases with radius values very close to zero. Instead we now using a tolerance value (Number.EPSILON).

### CODE REVIEW

File: src/extras/geometries/CylinderBufferGeometry.js [hunk 1]
```diff
@@ -153,7 +153,7 @@ function CylinderBufferGeometry( radiusTop, radiusBottom, height, radialSegments
 
 				// handle special case if radiusTop/radiusBottom is zero
 
-				if ( ( radiusTop === 0 && y === 0 ) || ( radiusBottom === 0 && y === heightSegments ) ) {
+				if ( ( radiusTop < Number.EPSILON && y === 0 ) || ( radiusBottom < Number.EPSILON && y === heightSegments ) ) {
```
Comment (WestLangley) [inline src/extras/geometries/CylinderBufferGeometry.js]: Do you need this special case? I'm not sure you do; just don't multiply by `radius` first?

**Comment by Mugen87** [comment]
@WestLangley Good point! What do you think about the last commit? A little bit more code but no special cases any more...

**Comment by WestLangley** [comment]
@Mugen87 Not tested, but something like:
```javascript
var u = x / radialSegments;

var theta = u * thetaLength + thetaStart;

var sinTheta = Math.sin( theta );
var cosTheta = Math.cos( theta );

// vertex
vertices.setXYZ( index, radius * sinTheta, - v * height + halfHeight, radius * cosTheta );

// normal
normal.set( sinTheta, tanTheta, cosTheta ).normalize(); // tanTheta should be renamed, IMO
normals.setXYZ( index, normal.x, normal.y, normal.z );
```

**Comment by Mugen87** [comment]
`tanTheta` is calculated in this way:

`var tanTheta = ( radiusBottom - radiusTop ) / height;`

Would be something like `incline` or `slope` a better name?

**Comment by WestLangley** [comment]
Doesn't matter. It is just not the same `theta`. I like `slope`. : - )

**Comment by Mugen87** [comment]
Done! 😊

**Comment by mrdoob** [comment]
@WestLangley all good?

**Comment by WestLangley** [comment]
OK. Thanks, @Mugen87.

**Comment by mrdoob** [comment]
Yays!

**Comment by 50Wliu** [comment]
Thanks everyone, I really appreciate it! :smile:

**Pull request merged**